app/models/manifest_source.rb

```
@@ -9,4 +9,22 @@ class ManifestSource < ActiveRecord::Base
   has_many :records
 
   validates :manifest, :source, presence: true
+  validates :manifest, uniqueness: { scope: :source }
+
+  def start!
+    return if success? && fetched_at && fetched_at > 3.hours.ago
+
+    # TODO: what do we do if it is pending, hmmmm
+    # If it is pending, don't make another request
+    update(status: :pending)
+    V2::DownloadManifestJob.perform_now(service, self)
+    update(status: :success, fetched_at: Time.zone.now)
+
+    # TODO: handle error
+    # TODO: start downloading records from VBMS and save to S3
+  end
+
+  def service
+    source == "VBMS" ? VBMSService : VVAService
```

Can we use a case statement here? Not clear we will ever expand to other services, but if we do it'd be a nice pattern to follow. Also if it's not either VVA or VBMS we probably want to throw an error.

---

Yes!